src/pages/messages/messages.ts

```diff
@@ -57,40 +59,135 @@ export class MessagesPage implements OnInit, OnDestroy {
-          messages.forEach((message: Message) => {
-            message.ownership = this.senderId == message.senderId ? 'mine' : 'other';
-          });
-
-          return messages;
-        });
-      });
-    });
-
     this.autoScroller = this.autoScroll();
+    this.subscribeMessages();
+
+    // Get total messages count in database so we can have an indication of when to
+    // stop the auto-subscriber
+    MeteorObservable.call('countMessages').subscribe((messagesCount: number) => {
+      // Note that the 'scroller' element is being created dynamically by an Ionic
+      // component and therefore the event listener can't be registered directly
+      // from the view
+      let scrollListener = this.onScroll.bind(this);
+      this.scroller.addEventListener('scroll', scrollListener);
+      this.autoRemoveScrollListener(messagesCount, scrollListener);
+    });
   }
 
   ngOnDestroy() {
     this.autoScroller.disconnect();
   }
 
-  onInputKeypress({keyCode}: KeyboardEvent): void {
+  onInputKeypress({ keyCode }: KeyboardEvent): void {
     if (keyCode == 13) {
       this.sendMessage();
     }
   }
 
+  onScroll(e): void {
+    // Unless we're at the top of the messages page
+    if (!this.scroller.scrollTop) {
+      this.subscribeMessages();
+    }
+  }
+
+  showOptions(): void {
+    const popover = this.popoverCtrl.create(MessagesOptionsComponent, {
+      chat: this.selectedChat
+    }, {
+      // Will be used as a CSS selector in our style-sheet
+      cssClass: 'options-popover'
+    });
+
+    popover.present();
+  }
+
+  // Subscribes to the relevant set of messages
+  subscribeMessages(): Subscription {
+    // Prohibit parallel subscriptions
+    if (this.loadingMessages) return;
+    // A flag which indicates if there's a subscription in process
+    this.loadingMessages = true;
+    // A custom offset to be used to re-adjust the scrolling position once
+    // new dataset is fetched
+    this.scrollOffset = this.scroller.scrollHeight;
+
+    return MeteorObservable.subscribe('messages',
+      this.selectedChat._id,
+      ++this.messagesBatchCounter
+    ).subscribe(() => {
+      // Keep tracking changes in the dataset and re-render the view
+      if (!this.messagesComputation) this.messagesComputation = this.autorunMessages();
+      // Allow incoming subscription requests
+      this.loadingMessages = false;
+    });
+  }
+
+  // Removes the scroll listener once all messages from the past were fetched
+  autoRemoveScrollListener(
+    messagesCount: number,
+    scrollListener: EventListener
+  ): Subscription {
+    return MeteorObservable.autorun().subscribe(() => {
+      if (messagesCount != Messages.collection.find().count()) return;
+      this.scroller.removeEventListener('scroll', scrollListener);
+    });
+  }
+
+  // Detects changes in the messages dataset and re-renders the view
+  autorunMessages(): Subscription {
+    return MeteorObservable.autorun().subscribe(() => {
```

Comment:
The `autorun` is used to re-render the view in-case there are incoming messages from the receiver. What do you mean by "use the collection"? How exactly do you want me to use it?